fix: update get offer endpoint logic to use limit and offset

HijoDeHonor opened the pull request:
- Changed logic in the GET endpoint to only return offers within specified limit and offset
- Adjusted unit tests to reflect the new logic
- Updated mocks in tests to align with the new logic
- Enhanced error handling and validation

HijoDeHonor committed:
fix: update get offer endpoint logic to use limit and offset

- Changed logic in the GET endpoint to only return offers within specified limit and offset
- Adjusted unit tests to reflect the new logic
- Updated mocks in tests to align with the new logic
- Enhanced error handling and validation

HijoDeHonor committed:
refactor: update get offer endpoint to return offers based on page

- Changed logic in the GET endpoint to return offers based on a page number
- Adjusted unit tests to reflect the new pagination logic

PiewDev commented on src/offer/offerRepository.js:
```diff
@@ -1,78 +1,100 @@
-     o.deleted = FALSE
-   ORDER BY 
-     o.id, offer_item_name, request_item_name;`
+      const limit = parseInt(process.env.PAGESIZE);
```
Don't use an env for the page size, use a constant

PiewDev commented on src/offer/offerService.js:
```diff
@@ -1,14 +1,17 @@
-  getOffers = async () => {
-    const offers = await this.offerRepository.getOffers();
+  getOffers = async (page) => {
+    if (!page) {
```
Maybe here it would not throw an error but would use page 1 by default

PiewDev reviewed (changes requested):
There are still small adjustments to make, a new pr is not necessary.

HijoDeHonor committed:
fix: change variable source and use default value

- Changed the source of a variable for better reliability
- Fixed an error by implementing a default value
